pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -116,6 +123,26 @@ type BGPStandardCommunity string
 // +kubebuilder:validation:Pattern=`^([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5])$`
 type BGPLargeCommunity string
 
+// BGPExtendedCommunity type represents a value of the BGP Extended Communities Attribute (RFC 4360).
+type BGPExtendedCommunity struct {
+	// SubType indicates a particular sub-type of extended community.
+	//
+	// +kubebuilder:validation:Enum=RouteTarget;RouteOrigin;LinkBandwidth
+	// +kubebuilder:validation:Required
+	SubType string `json:"subType"`
+
+	// Value holds the actual value of the extended community of the given sub-type.
+	// It can be represented in one the following formats:
+	// - Two-Octet AS Specific Extended Community: 2-byte decimal number + 4-byte decimal number separated by a colon.
+	// - Four-Octet AS-Specific Extended Community: two 2-byte decimal numbers separated by a dot + 2-byte decimal number separated by a colon.
+	// - IPv4-Address-Specific Extended Community: IPv4 address + 2-byte decimal number separated by a colon.
+	// - IPv6-Address-Specific Extended Community: IPv6 address + 2-byte decimal number separated by a colon.
```

Comment:
I found that this list doesn't appear well in the `kubectl explain`. Do we have any way to preserve line breaks? 

```
$ kc explain bgpp.spec.virtualRouters.neighbors.advertisedPathAttributes.communities.extended
KIND:     CiliumBGPPeeringPolicy
VERSION:  cilium.io/v2alpha1

RESOURCE: extended <[]Object>

DESCRIPTION:
     Extended holds a list of the BGP Extended Community Attribute (RFC 4360)
     values.

     BGPExtendedCommunity type represents a value of the BGP Extended
     Communities Attribute (RFC 4360).

FIELDS:
   subType	<string> -required-
     SubType indicates a particular sub-type of extended community.

   value	<string> -required-
     Value holds the actual value of the extended community of the given
     sub-type. It can be represented in one the following formats: - Two-Octet
     AS Specific Extended Community: 2-byte decimal number + 4-byte decimal
     number separated by a colon. - Four-Octet AS-Specific Extended Community:
     two 2-byte decimal numbers separated by a dot + 2-byte decimal number
     separated by a colon. - IPv4-Address-Specific Extended Community: IPv4
     address + 2-byte decimal number separated by a colon. -
     IPv6-Address-Specific Extended Community: IPv6 address + 2-byte decimal
     number separated by a colon.
```

Reply:
Good point, just figured out that adding a blank line between 2 lines adds a line break, so with that I was able to get this result:

```
GROUP:      cilium.io
KIND:       CiliumBGPPeeringPolicy
VERSION:    v2alpha1

FIELD: extended <[]Object>

DESCRIPTION:
    Extended holds a list of the BGP Extended Community Attribute (RFC 4360)
    values.
    BGPExtendedCommunity type represents a value of the BGP Extended Communities
    Attribute (RFC 4360).
    
FIELDS:
  subType	<string> -required-
    SubType indicates a particular sub-type of extended community.

  value	<string> -required-
    Value holds the actual value of the extended community of the given
    sub-type. It can be represented in one the following formats: 
     - Two-Octet AS Specific Extended Community: 2-byte decimal number + 4-byte
    decimal number separated by a colon. 
     - Four-Octet AS-Specific Extended Community: two 2-byte decimal numbers
    separated by a dot + 2-byte decimal number separated by a colon. 
     - IPv4-Address-Specific Extended Community: IPv4 address + 2-byte decimal
    number separated by a colon. 
     - IPv6-Address-Specific Extended Community: IPv6 address + 2-byte decimal
    number separated by a colon.
```